src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
+const ANIMAL_HEALTH = 100;
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = ANIMAL_HEALTH;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  static killedAnimal(victim) {
+    Animal.alive = Animal.alive.filter(animal => animal !== victim);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
```

Comment:
I agree with your logic, it is better to toggle hidden in such cases, but here in accordance with [checklist](https://github.com/mate-academy/js_herbivores_and_carnivores/blob/master/checklist.md) we should just hide, `this.hidden = true`, method name should reflect it's logic, `toggleHidden` for example can contain your logic, but here we should not change method's name